chainer/distributions/normal.py

```diff
@@ -0,0 +1,239 @@
+        super(Normal, self).__init__()
+        if isinstance(loc, chainer.Variable):
+            self.loc = loc
+        else:
+            self.loc = chainer.Variable(loc)
+        if isinstance(scale, chainer.Variable):
+            self.scale = scale
+        else:
+            self.scale = chainer.Variable(scale)
+
+    def __copy__(self):
+        return self._copy_to(Normal(self.loc, self.scale))
+
+    @property
+    def batch_shape(self):
+        return self.loc.shape
+
+    def cdf(self, x):
+        """Returns Cumulative Distribution Function for a input variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing Cumulative
+            Distribution Function.
+
+        """
+        return 0.5 * (1. + erf.erf((
+            x - broadcast.broadcast_to(self.loc, x.shape))
+            / (2 ** 0.5 * broadcast.broadcast_to(self.scale, x.shape))))
+
+    @property
+    def entropy(self):
+        """Returns entropy.
+
+        Returns:
+            Output Variable representing entropy.
+
+        """
+        return exponential.log(self.scale) \
+            + 0.5 * numpy.log(2 * numpy.pi * numpy.e)
+
+    @property
+    def event_shape(self):
+        return ()
+
+    def icdf(self, x):
+        """Returns Inverse Cumulative Distribution Function for a input Variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing Inverse Cumulative
+            Distribution Function.
+
+        """
+        return erfinv.erfinv(2. * x - 1.) \
+            * (2 ** 0.5) * broadcast.broadcast_to(self.scale, x.shape) \
+            + broadcast.broadcast_to(self.loc, x.shape)
+
+    @property
+    def _is_gpu(self):
+        return isinstance(self.loc.data, cuda.ndarray)
+
+    def log_cdf(self, x):
+        """Returns logarithm of Cumulative Distribution Function for a input Variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
+            variable.
+
+        Returns:
+            ~chainer.Variable: Output variable representing logarithm of
+            Cumulative Distribution Function.
+
+        """
+        return exponential.log(self.cdf(x))
+
+    def log_prob(self, x):
+        """Returns logarithm logarithm of probability for a input variable.
+
+        Args:
+            x: Input variable representing a random variable.
+
+        Returns:
+            Output variable representing logarithm of probability.
+
+        """
+        return - 0.5 * numpy.log(2 * numpy.pi) \
```

Comment:
How about precomputing `-0.5 * numpy.log(2 * numpy.pi)` at the global scope and just using it here? It is also better ot use `math` instead of `numpy` for it (because numpy is slow for scalar computations).